In [2]:
import pandas as pd

### Exploratory analysis of worldwide deaths (1990-2019),Orginal Data Source: https://ourworldindata.org/causes-of-death


In [ ]:
#data: Reading worldwide deaths data from azure blob storage via generatig SAS URL
data=pd.read_csv('https://imtprojectstorage.blob.core.windows.net/projectdatafiles/annual-number-of-deaths-by-cause.csv?sp=r&st=2022-01-25T02:47:54Z&se=2022-06-30T09:47:54Z&spr=https&sv=2020-08-04&sr=b&sig=GQEJ0Yz%2Bj1lg%2FgLlo36jglAAnyWfm8qGh5nSc8MTHeQ%3D')

In [4]:
data.head()

Entity Code  Year Number of executions (Amnesty International)  \
0  Afghanistan  AFG  2007                                           15   
1  Afghanistan  AFG  2008                                           17   
2  Afghanistan  AFG  2009                                            0   
3  Afghanistan  AFG  2011                                            2   
4  Afghanistan  AFG  2012                                           14   

   Deaths - Meningitis - Sex: Both - Age: All Ages (Number)  \
0                                        2932.558591          
1                                        2730.846137          
2                                        2460.186423          
3                                        2326.767573          
4                                        2254.492968          

   Deaths - Neoplasms - Sex: Both - Age: All Ages (Number)  \
0                                        15924.78254         
1                                        16147.83903         
2                                        16382.55855         
3                                        17093.51567         
4                                        17521.93019         

   Deaths - Fire, heat, and hot substances - Sex: Both - Age: All Ages (Number)  \
0                                         481.219817                              
1                                         462.180576                              
2                                         448.300264                              
3                                         447.884353                              
4                                         445.226412                              

   Deaths - Malaria - Sex: Both - Age: All Ages (Number)  \
0                                         392.942607       
1                                         254.703403       
2                                         239.035635       
3                                         390.060248       
4                                          93.894201       

   Deaths - Drowning - Sex: Both - Age: All Ages (Number)  \
0                                        2126.788535        
1                                        1973.375893        
2                                        1852.155239        
3                                        1774.968476        
4                                        1715.864705        

   Deaths - Interpersonal violence - Sex: Both - Age: All Ages (Number)  ...  \
0                                        3657.205757                     ...   
1                                        3784.702895                     ...   
2                                        3874.099828                     ...   
3                                        4169.873038                     ...   
4                                        4244.671108                     ...   

   Deaths - Protein-energy malnutrition - Sex: Both - Age: All Ages (Number)  \
0                                        2438.567035                           
1                                        2230.538625                           
2                                        1997.531216                           
3                                        1805.467089                           
4                                        1666.898963                           

   Terrorism (deaths)  \
0              1199.0   
1              1092.0   
2              1065.0   
3              1525.0   
4              3521.0   

   Deaths - Cardiovascular diseases - Sex: Both - Age: All Ages (Number)  \
0                                        53962.25186                       
1                                        54050.64420                       
2                                        53964.19556                       
3                                        54347.26659                       
4                                        54868.12037                       

   Deaths - Ch

In [5]:
data.shape

(8590, 36)

In [6]:
#Checking the column names and datatypes
data.dtypes

Entity                                                                                       object
Code                                                                                         object
Year                                                                                          int64
Number of executions (Amnesty International)                                                 object
Deaths - Meningitis - Sex: Both - Age: All Ages (Number)                                    float64
Deaths - Neoplasms - Sex: Both - Age: All Ages (Number)                                     float64
Deaths - Fire, heat, and hot substances - Sex: Both - Age: All Ages (Number)                float64
Deaths - Malaria - Sex: Both - Age: All Ages (Number)                                       float64
Deaths - Drowning - Sex: Both - Age: All Ages (Number)                                      float64
Deaths - Interpersonal violence - Sex: Both - Age: All Ages (Number)                        float64


In [33]:
data.nunique(axis=0)

Entity                                                                                       304
Code                                                                                         217
Year                                                                                          30
Number of executions (Amnesty International)                                                  81
Deaths - Meningitis - Sex: Both - Age: All Ages (Number)                                    8160
Deaths - Neoplasms - Sex: Both - Age: All Ages (Number)                                     8160
Deaths - Fire, heat, and hot substances - Sex: Both - Age: All Ages (Number)                8160
Deaths - Malaria - Sex: Both - Age: All Ages (Number)                                       4201
Deaths - Drowning - Sex: Both - Age: All Ages (Number)                                      8160
Deaths - Interpersonal violence - Sex: Both - Age: All Ages (Number)                        8160
Deaths - HIV/AIDS - Sex: Both 

In [8]:
#Checking the unique values for entity in the dataset
data['Entity'].unique()

array(['Afghanistan', 'Africa', 'African Region', 'African Union',
       'Albania', 'Algeria', 'America', 'American Samoa',
       'Andean Latin America', 'Andorra', 'Angola', 'Antigua and Barbuda',
       'Argentina', 'Armenia', 'Asia', 'Australasia',
       'Australasia & Oceania', 'Australia', 'Austria', 'Azerbaijan',
       'Bahamas', 'Bahrain', 'Bangladesh', 'Barbados', 'Belarus',
       'Belgium', 'Belize', 'Benin', 'Bermuda', 'Bhutan', 'Bolivia',
       'Bosnia and Herzegovina', 'Bosnia-Herzegovina', 'Botswana',
       'Brazil', 'Brunei', 'Bulgaria', 'Burkina Faso', 'Burundi',
       'Cambodia', 'Cameroon', 'Canada', 'Cape Verde', 'Caribbean',
       'Central African Republic', 'Central America & Caribbean',
       'Central Asia', 'Central Europe',
       'Central Europe, Eastern Europe, and Central Asia',
       'Central Latin America', 'Central sub-Saharan Africa', 'Chad',
       'Chile', 'China', 'Colombia', 'Commonwealth',
       'Commonwealth High Income', 'Commonwealth Lo

#### Column entity has 304 unique values which includes countries, income group, region, world values

In [9]:
#Checking if the dataset has missing values
data.isnull().sum()

Entity                                                                                         0
Code                                                                                        2384
Year                                                                                           0
Number of executions (Amnesty International)                                                8323
Deaths - Meningitis - Sex: Both - Age: All Ages (Number)                                     370
Deaths - Neoplasms - Sex: Both - Age: All Ages (Number)                                      370
Deaths - Fire, heat, and hot substances - Sex: Both - Age: All Ages (Number)                 370
Deaths - Malaria - Sex: Both - Age: All Ages (Number)                                        370
Deaths - Drowning - Sex: Both - Age: All Ages (Number)                                       370
Deaths - Interpersonal violence - Sex: Both - Age: All Ages (Number)                         370
Deaths - HIV/AIDS - Sex: Both 

In [10]:
#data2 : Analyzing transformed file of worldwide deaths 
data2= pd.read_csv('https://imtprojectstorage.blob.core.windows.net/projectdatafilestransformed/transformed-annual-number-of-deaths-by-cause_Country.csv?sp=r&st=2022-01-31T19:34:28Z&se=2022-02-01T03:34:28Z&spr=https&sv=2020-08-04&sr=b&sig=mdF6YMofBkSqMbv8PD1ShLLehV55hJgYitrXh59R%2BnI%3D')

In [11]:
data2.head()

Country  Year                    DeathReasons  TotalDeaths
0  Afghanistan  2007                      Executions         15.0
1  Afghanistan  2007                      Meningitis       2932.0
2  Afghanistan  2007                       Neoplasms      15924.0
3  Afghanistan  2007  Fire, heat, and hot substances        481.0
4  Afghanistan  2007                         Malaria        392.0

In [12]:
data2.info

<bound method DataFrame.info of             Country  Year                                DeathReasons  \
0       Afghanistan  2007                                  Executions   
1       Afghanistan  2007                                  Meningitis   
2       Afghanistan  2007                                   Neoplasms   
3       Afghanistan  2007              Fire, heat, and hot substances   
4       Afghanistan  2007                                     Malaria   
...             ...   ...                                         ...   
182155     Zimbabwe  2019  Cirrhosis and other chronic liver diseases   
182156     Zimbabwe  2019                          Digestive diseases   
182157     Zimbabwe  2019                             Acute hepatitis   
182158     Zimbabwe  2019     Alzheimer's disease and other dementias   
182159     Zimbabwe  2019                         Parkinson's disease   

        TotalDeaths  
0              15.0  
1            2932.0  
2           15924.0  
3  

In [13]:
data2.nunique(axis=0)

Country           184
Year               30
DeathReasons       33
TotalDeaths     24077
dtype: int64

#### death data for 184 countries is available

In [14]:
#Creating a list of countries for which death records is available in tranformed dataset
countrylist1= data2['Country'].unique()
print(countrylist1)

['Afghanistan' 'Albania' 'Algeria' 'Andorra' 'Angola'
 'Antigua and Barbuda' 'Argentina' 'Armenia' 'Australia' 'Austria'
 'Azerbaijan' 'Bahamas' 'Bahrain' 'Bangladesh' 'Barbados' 'Belarus'
 'Belgium' 'Belize' 'Benin' 'Bhutan' 'Bolivia' 'Bosnia and Herzegovina'
 'Botswana' 'Brazil' 'Brunei' 'Bulgaria' 'Burkina Faso' 'Burundi'
 'Cambodia' 'Cameroon' 'Canada' 'Central African Republic' 'Chad' 'Chile'
 'China' 'Colombia' 'Comoros' 'Costa Rica' 'Croatia' 'Cuba' 'Cyprus'
 'Denmark' 'Djibouti' 'Dominica' 'Dominican Republic' 'Ecuador' 'Egypt'
 'El Salvador' 'Equatorial Guinea' 'Eritrea' 'Estonia' 'Ethiopia' 'Fiji'
 'Finland' 'France' 'Gabon' 'Gambia' 'Georgia' 'Germany' 'Ghana' 'Greece'
 'Grenada' 'Guatemala' 'Guinea' 'Guinea-Bissau' 'Guyana' 'Haiti'
 'Honduras' 'Hungary' 'Iceland' 'India' 'Indonesia' 'Iran' 'Iraq'
 'Ireland' 'Israel' 'Italy' 'Jamaica' 'Japan' 'Jordan' 'Kazakhstan'
 'Kenya' 'Kiribati' 'Kuwait' 'Kyrgyzstan' 'Laos' 'Latvia' 'Lebanon'
 'Lesotho' 'Liberia' 'Libya' 'Lithuania' 'Lu

In [15]:
#Reading countries from country list
datacountrylist= pd.read_csv('https://imtprojectstorage.blob.core.windows.net/projectdatafiles/CountryList.csv?sp=r&st=2022-01-31T19:36:01Z&se=2023-05-01T02:36:01Z&spr=https&sv=2020-08-04&sr=b&sig=xyTEnJklZjv3jAZQ218PSgtwdYskYmG7jtCL7Rym%2Baw%3D')

In [16]:
datacountrylist.head()

Id      Country
0   1  Afghanistan
1   2      Albania
2   3      Algeria
3   4      Andorra
4   5       Angola

In [17]:
datacountrylist.nunique(axis=0)

Id         195
Country    195
dtype: int64

In [18]:
#Creating a list for the total of 195 countries
countrylist2= datacountrylist['Country'].unique()
print(countrylist2)

['Afghanistan' 'Albania' 'Algeria' 'Andorra' 'Angola'
 'Antigua and Barbuda' 'Argentina' 'Armenia' 'Australia' 'Austria'
 'Azerbaijan' 'Bahamas' 'Bahrain' 'Bangladesh' 'Barbados' 'Belarus'
 'Belgium' 'Belize' 'Benin' 'Bhutan' 'Bolivia' 'Bosnia and Herzegovina'
 'Botswana' 'Brazil' 'Brunei' 'Bulgaria' 'Burkina Faso' 'Burundi'
 "Côte d'Ivoire" 'Cabo Verde' 'Cambodia' 'Cameroon' 'Canada'
 'Central African Republic' 'Chad' 'Chile' 'China' 'Colombia' 'Comoros'
 'Congo (Congo-Brazzaville)' 'Costa Rica' 'Croatia' 'Cuba' 'Cyprus'
 'Czechia (Czech Republic)' 'Democratic Republic of the Congo' 'Denmark'
 'Djibouti' 'Dominica' 'Dominican Republic' 'Ecuador' 'Egypt'
 'El Salvador' 'Equatorial Guinea' 'Eritrea' 'Estonia'
 'Eswatini (fmr. "Swaziland")' 'Ethiopia' 'Fiji' 'Finland' 'France'
 'Gabon' 'Gambia' 'Georgia' 'Germany' 'Ghana' 'Greece' 'Grenada'
 'Guatemala' 'Guinea' 'Guinea-Bissau' 'Guyana' 'Haiti' 'Holy See'
 'Honduras' 'Hungary' 'Iceland' 'India' 'Indonesia' 'Iran' 'Iraq'
 'Ireland' 'Israe

In [19]:
#Converting list of countries to dataframe
df2 = pd.DataFrame (countrylist2, columns = ['Country'])

In [20]:
#Checking which all countries out of 195 countries are not available in the tranformed dataset
df2[~df2['Country'].isin(countrylist1)]

Country
28                      Côte d'Ivoire
29                         Cabo Verde
39          Congo (Congo-Brazzaville)
44           Czechia (Czech Republic)
45   Democratic Republic of the Congo
56        Eswatini (fmr. "Swaziland")
73                           Holy See
98                      Liechtenstein
111                        Micronesia
133                   Palestine State
174                       Timor-Leste

#### Exploratory Analysis of Population Data 1960-2020 Original Data Source:https://data.worldbank.org/indicator/SP.POP.TOTL?end=2020&start=1990

In [21]:
#data3 : Population Dataset
data3= pd.read_csv('https://imtprojectstorage.blob.core.windows.net/projectdatafiles/Countries%20by%20Population.csv?sp=r&st=2022-01-31T19:36:57Z&se=2023-08-01T02:36:57Z&spr=https&sv=2020-08-04&sr=b&sig=GHT0SjJJOidJ%2BODjX%2FQfmddy%2Fpl09%2FRp5CaVEXHZgE0%3D')

In [22]:
data3.shape

(266, 65)

In [23]:
data3.dtypes

Country Name       object
Country Code       object
Indicator Name     object
Indicator Code     object
1960              float64
                   ...   
2016              float64
2017              float64
2018              float64
2019              float64
2020              float64
Length: 65, dtype: object

In [24]:
data3.nunique(axis=0)

Country Name      266
Country Code      266
Indicator Name      1
Indicator Code      1
1960              260
                 ... 
2016              262
2017              262
2018              262
2019              262
2020              262
Length: 65, dtype: int64

In [25]:
data3.isnull().sum()

Country Name      0
Country Code      0
Indicator Name    0
Indicator Code    0
1960              2
                 ..
2016              2
2017              2
2018              2
2019              2
2020              2
Length: 65, dtype: int64

In [26]:
data3['Country Name'].unique()

array(['Aruba', 'Africa Eastern and Southern', 'Afghanistan',
       'Africa Western and Central', 'Angola', 'Albania', 'Andorra',
       'Arab World', 'United Arab Emirates', 'Argentina', 'Armenia',
       'American Samoa', 'Antigua and Barbuda', 'Australia', 'Austria',
       'Azerbaijan', 'Burundi', 'Belgium', 'Benin', 'Burkina Faso',
       'Bangladesh', 'Bulgaria', 'Bahrain', 'Bahamas',
       'Bosnia and Herzegovina', 'Belarus', 'Belize', 'Bermuda',
       'Bolivia', 'Brazil', 'Barbados', 'Brunei Darussalam', 'Bhutan',
       'Botswana', 'Central African Republic', 'Canada',
       'Central Europe and the Baltics', 'Switzerland', 'Channel Islands',
       'Chile', 'China', "Cote d'Ivoire", 'Cameroon', 'Congo, Dem. Rep.',
       'Congo, Rep.', 'Colombia', 'Comoros', 'Cabo Verde', 'Costa Rica',
       'Caribbean small states', 'Cuba', 'Curacao', 'Cayman Islands',
       'Cyprus', 'Czech Republic', 'Germany', 'Djibouti', 'Dominica',
       'Denmark', 'Dominican Republic', 'Algeria',

#### Population datatset has not only countries but regions too

In [27]:
#data4: Analyzing tranformed Population Dataset after joining it with country list dataset
data4=pd.read_csv('https://imtprojectstorage.blob.core.windows.net/projectdatafilestransformed/transformed-country-by-population.csv?sp=r&st=2022-01-31T19:37:47Z&se=2024-01-01T03:37:47Z&spr=https&sv=2020-08-04&sr=b&sig=WW6tIdUmYAQnvkYaTq%2B5KH6jxEUfU%2FYSBXmQSm%2F8OLQ%3D')

In [28]:
data4.shape

(5430, 3)

In [29]:
data4.head

<bound method NDFrame.head of       Population      Country  Year
0     12412311.0  Afghanistan  1990
1     13299016.0  Afghanistan  1991
2     14485543.0  Afghanistan  1992
3     15816601.0  Afghanistan  1993
4     17075728.0  Afghanistan  1994
...          ...          ...   ...
5425  13814642.0     Zimbabwe  2015
5426  14030338.0     Zimbabwe  2016
5427  14236599.0     Zimbabwe  2017
5428  14438812.0     Zimbabwe  2018
5429  14645473.0     Zimbabwe  2019

[5430 rows x 3 columns]>

In [30]:
data4.nunique(axis=0)

Population    5415
Country        181
Year            30
dtype: int64

#### Population data for 181 countries is fetched 